azurerm/resource_arm_cosmosdb_mongo_collection.go

```diff
@@ -78,7 +80,7 @@ func resourceArmCosmosDbMongoCollection() *schema.Resource {
 			"throughput": {
 				Type:         schema.TypeInt,
 				Optional:     true,
-				Default:      400,
+				Default:      nil,
```

Comment:
I think this will technically be a breaking change? While i agree we should make the change i'm not sure we can until 2.0 where we break/change schema

Reply:
That is a good question. If I'm correct it should not break anyones code / resources.
Case 1 - user had no throughput configured for existing resource -> terraform won't re-apply the throughput meaning it won't override whatever is configured in azure.
Case 2 - user creates a new resource -> will be created with the azure default (which should be 1000)
Case 3 - user had throughput configured -> will use that value

In the past my team had quite some trouble with newly added parameters with default values. In some cases it destroyed resources for us. In this case setting the throughput to a default of 400 might scale down collections for users which had them previously configured in terraform, but scaled via azure directly. If users roll their terraform config out automatically this is dangerous.

Reply:
What do you think, @katbyte ?

Reply:
Documentation and testing showed 400 being the azure default - but I do agree that defaulting to `nil` here probably makes more sense

Reply:
👍I think your correct, at worst the user will just have a plan with no affect.